mitiq/zne/inference.py

```diff
@@ -1511,6 +1513,141 @@ def __eq__(self, other: Any) -> bool:
+                   the same length of scale_factors and the executor function
+                   must accept "shots" as a valid keyword argument.
+
+    Raises:
+        ValueError: If data is not consistent with the extrapolation model.
+    Note:
+        RichardsonFactory and LinearFactory are special cases of PolyFactory.
+    """
+
+    @staticmethod
+    def _exp_ansatz(
+        a: float,
+        b: float,
+        c: float,
+        scale_factor: float
+    ) -> float:
+        """
+        Calculates the expecation given a scale factor and model
+        parameters.
+
+        Args:
+            a: Model parameter.
+            b: Model parameter.
+            c: Model parameter.
+            scale_factors: The array of noise scale factors.
+
+        Returns:
+            The expected value according to the model parameters and
+            scale factor.
+        """
+
+        return a + b * np.exp(-1.0 * c * scale_factor)
+
+    @staticmethod
+    def extrapolate(
+        scale_factors: Sequence[float],
+        exp_values: Sequence[float],
+        full_output: bool = False,
+    ) -> Union[
+
+        float,
+    ]:
+        """Static method which evaluates an exponential extrapolation to the
+        zero-noise limit using Bayesian inference.
+
+        Args:
+            scale_factors: The array of noise scale factors.
+            exp_values: The array of expectation values.
+            full_output: If False (default), only the zero-noise limit is
+                returned. If True, additional information about the
+                extrapolated limit is returned too.
+        Returns:
+            zne_limit: The extrapolated zero-noise limit. If "full_output"
+                is False (default value), only this parameter is returned.
+            opt_params: The parameter array of the best fitting model.
+            zne_curve: The callable function which best fit the input data.
+                It maps a real noise scale factor to a real expectation value.
+                It is equal "zne_limit" when evaluated at zero.
+
+        Note:
+            This method computes the zero-noise limit only from the information
+            contained in the input arguments. To extrapolate from the internal
+            data of an instantiated Factory object, the bound method
+            ".reduce()" should be called instead.
+        """
+        with pm.Model():
+            """
+            We assumme that the priors for the model parameters is a uniform
+            distribution with an upper limit 1 and a lower limit 0,
+            except for b with a lower limit -1.
+            """
+            a = pm.Uniform('a', 0, 1)
+            b = pm.Uniform('b', -1, 1)
+            c = pm.Uniform('c', 0, 1)
+
+            # Noise parameter
+            eps = pm.Uniform('eps', 0, 0.5)
+
+            pm.Normal(
+                'expval',
+                mu=ExpBayesFactory._exp_ansatz(a, b, c, scale_factors),
+                sd=eps,
```

Comment:
It looks like this takes an additive Gaussian noise model, and treats the standard deviation of each point in that model as a free variable to be learned. The variance is completely determined, however, by knowing 𝑎, 𝑏, 𝑐, and the number of shots, as the binomial model being approximated by this normal distribution is a single-parameter model. In particular, 𝑛 ~ Bin(𝑁, 𝑝) is approximated by 𝑛 ~ N(𝑁𝑝, 𝑁𝑝(1 − 𝑝)) (a normal distribution with mean 𝑁𝑝 and variance 𝑁𝑝(1 − 𝑝)) .

Thus, using the number of shots, you can eliminate `eps` as a model parameter, and can improve the quality of your inference. One way to see that is that since the amount of noise is a determinstic function of 𝑎, 𝑏, and 𝑐, learning how much noise there is also informs about the model parameters of interest; you lose that information if you represent the variance as a nuisance parameter.

Reply:
That is a really good point, will adjust.

Reply:
Unfortunately at the moment in Mitiq we do not assume that the number of shots are given (these are optional) or that the raw counts are given. Moreover we also don't assume that observables are binomials, e.g. the observable could be a linear combination of terms  e.g.  <Z_a - Z_b>, it could even be the full Hamiltonian of a system. What the user of Mitiq gives us is just a black-box function which given a circuit returns a real expectation value of _something_ .

So I think that for the moment we need to guess eps and we could also put it as an argument of the class with a guessed default value.